### PR TITLE
Create text_detection_demo.py

### PR DESCRIPTION
**Pull request by banderlog**
Python demo for text-detection-0001.

### CODE REVIEW

**Comment by snosov1** [comment]
@Ilya-Krylov please, review

File: demos/python_demos/text_detection_demo.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        for box in self.bboxes:
+            cv2.drawContours(img_tmp, [box], 0, (0,0,255), 2)
+        cv2.imshow('Detected text', img_tmp)
+        cv2.waitKey(0)
+        if cv2.waitKey():
+            cv2.destroyAllWindows()
+
+
+def main():
+    td = cv2.dnn.readNet(args.model_path + '/text-detection-0001.xml', args.model_path + '/text-detection-0001.bin')
```
Comment (Ilya-Krylov) [inline demos/python_demos/text_detection_demo.py]: model_path should point to an xml file as it is made in c++ sample. Moreover text-detection-0001 model will be updated by better model, so it is no very well to hard-code it's name.

File: demos/python_demos/text_detection_demo.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        out += a_max
+
+        if return_sign:
+            return out, sgn
+        else:
+            return out
+
+    def _set_pixel_scores(self, pixel_scores):
+        "get softmaxed properly shaped pixel scores"
+        tmp = np.transpose(pixel_scores, (0,2,3,1))
```
[the remark this answers is not in the document]
Comment (banderlog) [reply demos/python_demos/text_detection_demo.py]: https://github.com/opencv/open_model_zoo/pull/36/commits/a4e687debcde43c76b65fc2ba963f6fc56adf6e6

File: demos/python_demos/text_detection_demo.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        return self._softmax(tmp, axis=-1)[0, :, :, 1]
+
+    def _set_link_scores(self, link_scores):
+        "get softmaxed properly shaped links scores"
+        tmp = np.transpose(link_scores, (0,2,3,1))
```
[the remark this answers is not in the document]
Comment (banderlog) [reply demos/python_demos/text_detection_demo.py]: Done in https://github.com/opencv/open_model_zoo/pull/36/commits/a4e687debcde43c76b65fc2ba963f6fc56adf6e6

File: demos/python_demos/text_detection_demo.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+                if link_value and pixel_cls:
+                    self._join(point, (ny, nx))
+
+        self._get_all()
+        self._mask_to_bboxes()
+
+    def plot_result(self, image):
+        img_tmp = image.copy()
+        for box in self.bboxes:
+            cv2.drawContours(img_tmp, [box], 0, (0,0,255), 2)
```
[the remark this answers is not in the document]
Comment (banderlog) [reply demos/python_demos/text_detection_demo.py]: Done in https://github.com/opencv/open_model_zoo/pull/36/commits/a4e687debcde43c76b65fc2ba963f6fc56adf6e6

File: demos/python_demos/text_detection_demo.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+        cv2.imshow('Detected text', img_tmp)
+        cv2.waitKey(0)
+        if cv2.waitKey():
+            cv2.destroyAllWindows()
+
+
+def main():
+    td = cv2.dnn.readNet(args.model_path + '/text-detection-0001.xml', args.model_path + '/text-detection-0001.bin')
+    img = cv2.imread(args.image_path)
+    blob = cv2.dnn.blobFromImage(img, 1, (1280,768))
```
[the remark this answers is not in the document]
Comment (banderlog) [reply demos/python_demos/text_detection_demo.py]: Done in https://github.com/opencv/open_model_zoo/pull/36/commits/a4e687debcde43c76b65fc2ba963f6fc56adf6e6

File: demos/python_demos/text_detection_demo.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (banderlog) [reply demos/python_demos/text_detection_demo.py]: Done in https://github.com/opencv/open_model_zoo/pull/36/commits/a4e687debcde43c76b65fc2ba963f6fc56adf6e6

File: demos/python_demos/text_detection_demo.py [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+        self.pixel_scores = self._set_pixel_scores(pixel_scores)
+        self.link_scores = self._set_link_scores(link_scores)
+
+        if four_neighbours:
+            self._get_neighbours = self._get_neighbours_4
+        else:
+            self._get_neighbours = self._get_neighbours_8
+
+        if pixel_conf_threshold is None:
+            self.pixel_conf_threshold = 0.75
```
[the remark this answers is not in the document]
Comment (banderlog) [reply demos/python_demos/text_detection_demo.py]: Done in https://github.com/opencv/open_model_zoo/pull/36/commits/a4e687debcde43c76b65fc2ba963f6fc56adf6e6

**Comment by banderlog** [comment]
I am worried, I should not have clicked "resolve conversation"?

**Comment by snosov1** [comment]
Jenkins please retry a build

**Comment by Ilya-Krylov** [comment]
@snosov1 what about merging this stuff?

**Comment by banderlog** [comment]
Yay!

**Pull request merged**